SUPPORTING_NEW_SERVICES.md

```diff
@@ -0,0 +1,108 @@
+* Paging: Cloud services often expose page-based listing using page tokens.  The [`Page`](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-core/src/main/java/com/google/cloud/Page.java) interface should be used for page-based listing.  A `Page` contains an iterator over results in that page, as well as methods to get the next page and all results in future pages.  `Page` requires a `NextPageFetcher` implementation (see the `NextPageFetcher` interface in [`PageImpl`](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-core/src/main/java/com/google/cloud/PageImpl.java)).  This implementation should delegate constructing request options to the `nextRequestOptions` method in `PageImpl`.
+
+* Exception handling: we have the notion of retryable versus non-retryable operations.  These are encapsulated by `BaseServiceException`.  Retryable error codes should be listed in the service's subclass of `BaseServiceException`.  An operation should be considered retryable if it makes sense to retry (e.g. if there was a transient service error, not a fundamentally invalid request) and if the operation that triggered the exception is idempotent.  Exceptions also contain information regarding whether the service rejected the request, meaning the operation was not applied.  The `BaseServiceException` subclass should also provide methods to translate the exceptions given by the underlying autogeneraged client library.  The [`ExceptionHandler`](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-core/src/main/java/com/google/cloud/ExceptionHandler.java) class intercepts and retries RPC calls when retryable exceptions are encountered.  Note that some exceptions are masked in the SPI layer.  For example, `get` and `delete` operations often return "404 Not Found" if the resource doesn't exist.  Instead of throwing an exception in these cases, we often return `null` for `get` and `false` for `delete`.
+
+* Batching: The [`BatchResult`](https://github.com/GoogleCloudPlatform/gcloud-java/blob/dns-alpha-batch/gcloud-java-core/src/main/java/com/google/cloud/BatchResult.java) class provides a simple way for users to combine RPC calls for performance enhancement.  API for services that support batching should implement a batch class that contains methods similar to the methods in the `Service.java` subclass, but instead return a subclass of `BatchResult`.  Also provide an SPI-layer class to collect batch requests and submit the batch.
+
+Notes/reminders:
+* API layer classes should be located in the package `com.google.cloud.servicename`, where "servicename" corresponds to the name of the Cloud service.
+* Override the `ServiceOptions.defaultRetryParams()` method in your service's options class to align with the Service Level Agreement (SLA) given by the underlying service.  See #857 and #860 for context.
```

Comment:
Regarding overriding ServiceOptions method, should we mention overriding `projectIdRequired` to return false when service does not require a project to be associated with specific project?